test_requests.py

```diff
@@ -1301,6 +1301,28 @@ def test_get_auth_from_url(self):
         assert username == percent_encoding_test_chars
         assert password == percent_encoding_test_chars
 
+    def test_requote_uri_with_unquoted_percents(self):
+        """Ensure we handle unquoted percent signs in redirects.
+
+        See: https://github.com/kennethreitz/requests/issues/2356
+        """
+        from requests.utils import requote_uri
+        bad_uri = ('http://style.shoedazzle.com/dmg/3AE3B8?dzcode=FBT'
+                   '&dzcontent=FBT_SDZ_CPM_Q414&pid=112768085&aid=285880402'
+                   '&cid=0&publisher=%ppublisher=!;&placement=%pplacement=!;')
+        quoted = ('http://style.shoedazzle.com/dmg/3AE3B8?dzcode=FBT'
+                  '&dzcontent=FBT_SDZ_CPM_Q414&pid=112768085&aid=285880402&cid'
+                  '=0&publisher=%25ppublisher=!;&placement=%25pplacement=!;')
```

Comment:
These URIs are massive so it makes it easy to miss what you're actually doing. Can we shorten them to a smaller case?